### DISCUSSION TITLE
Mpd Settings Issue

### DISCUSSION THREAD

**Discussion by Ondi78** [rern/rAudio#93] [Fixed Issues]
Hi rAudio Community,

When I disconnect my DAC (unplug usb cable), the sections "Output" and "Bit-perfect" disappear in the Mpd settings. After reconnect the DAC the sections don't reappear and nothing is streamed to the DAC after that. Reboots or reconnects don't fix it. But it seems, that the Pi and the DAC have a connection because there is a click noise on my speakers during reboot.

Currently my only solution is to reinstall the rAudio image.

Hardware: Pi 3B+
Software: rAudio1 20210108

Regards,
Ondi

**Comment by Roland-B** [comment]
@rern 

Same here, cannot test the fix for Pi4 for streaming Webradio as I cannot see/play from my my DAC or any other audio device anymore as the settings seems to have dissapeared after applying the update before todays one.

RolandB

**Comment by rern** [comment]
@Ondi78, @Roland-B,

Post your:
`mpd-data.sh`

**Reply by Roland-B** [reply]
@rern 

mpd-data.sh
/srv/http/bash/mpd-data.sh: line 32: Aaplayname: bad array subscript
/srv/http/bash/mpd-data.sh: line 32: Aname: bad array subscript
{ "devices" : [] , "active" : true , "asoundcard" : -1 , "audioaplayname" : "" , "audiooutput" : "" , "autoupdate" : false , "buffer" : false , "bufferval" : false , "bufferoutput" : false , "bufferoutputval" : false , "counts" : { "album": 0, "albumartist": 0, "artist": 0, "composer": 0, "date": 0, "genre": 0, "nas": 0, "sd": 0, "usb": 0, "song": 0, "webradio": 5 } , "crossfade" : false , "crossfadeval" : false , "custom" : false , "ffmpeg" : true , "normalization" : false , "reboot" : "" , "replaygain" : false , "replaygainval" : "" , "soxr" : false , "soxrval" : "" , "version" : "0.22.3-2" }

Quick note, did quick re-image and configured DAC and applied update and rebooted. As with @Ondi78 it only breaks after disconnecting the USB DAC and reconnecting and the "Output" section in MPD Settings dissapears completely, so I will reimage now and not disconnect the DAC and test the webradio fix and report back on that one. Thanks.

RolandB

**Comment by Ondi78** [comment]
@rern 
I can't reproduce the behavior after my last re-image (still using 20210108). It is playing on my DAC even after re-plug the usb. But I can't select it as output device. Headphones is always active. Strange...

Here my mpd-data:
{ "devices" : [{ "aplayname" : "bcm2835 Headphones" , "card" : 0 , "device" : 0 , "dop" : 0 , "hw" : "hw:0,0" , "hwmixer" : "Headphone" , "mixers" : 1 , "mixerdevices" : ["Headphone"] , "mixermanual" : false , "mixertype" : "none" , "name" : "On-board Headphones" },{ "aplayname" : "xCORE USB Audio 2.0" , "card" : 1 , "device" : 0 , "dop" : 0 , "hw" : "hw:1,0" , "hwmixer" : "xCORE USB Audio 2.0 " , "mixers" : 1 , "mixerdevices" : ["xCORE USB Audio 2.0 "] , "mixermanual" : false , "mixertype" : "none" , "name" : "xCORE USB Audio 2.0" }] , "active" : true , "asoundcard" : 0 , "audioaplayname" : "bcm2835 Headphones" , "audiooutput" : "On-board Headphones" , "autoupdate" : true , "buffer" : false , "bufferval" : false , "bufferoutput" : false , "bufferoutputval" : false , "counts" : { "album": 9, "albumartist": 8, "artist": 32, "composer": 0, "date": 3, "genre": 1, "nas": 0, "sd": 0, "usb": 1, "song": 133, "webradio": 4 } , "crossfade" : false , "crossfadeval" : false , "custom" : false , "ffmpeg" : false , "normalization" : false , "reboot" : "" , "replaygain" : false , "replaygainval" : "" , "soxr" : false , "soxrval" : "" , "version" : "0.22.3-2" }

**Comment by Ondi78** [comment]
Today, the problem with mpd settings and no sound on my DAC is back. Version 20210109.


mpd-data.sh: line 32: Aaplayname: bad array subscript
mpd-data.sh: line 32: Aname: bad array subscript
{ "devices" : [] , "active" : true , "asoundcard" : -1 , "audioaplayname" : "" , "audiooutput" : "" , "autoupdate" : false , "buffer" : false , "bufferval" : false , "bufferoutput" : false , "bufferoutputval" : false , "counts" : { "album": 36, "albumartist": 20, "artist": 67, "composer": 1, "date": 17, "genre": 6, "nas": 0, "sd": 0, "usb": 1, "song": 518, "webradio": 0 } , "crossfade" : false , "crossfadeval" : false , "custom" : false , "ffmpeg" : false , "normalization" : false , "reboot" : "" , "replaygain" : false , "replaygainval" : "" , "soxr" : false , "soxrval" : "" , "version" : "0.22.3-2" }

**Comment by Roland-B** [comment]
@rern 
@Ondi78 

Fixed for me with update 202101111. Cheers.

**Reply by Ondi78** [reply]
Same for me. Great! Thanks!
